docs/providers/provider-protocol-reference.md

```diff
@@ -4,183 +4,195 @@ category: 63e0c7cf7166bf006d3b4117
 
-## Imported functions
+We use [`fp-bindgen`](https://github.com/fiberplane/fp-bindgen) for generating the bindings themselves.
 
-These functions are usable as long as you import `fiberplane_pdk::prelude::*` in your library code. These functions exist to allow you to use OS-level features by asking the runtime for information.
+## Exported functions
 
-For the Fiberplane Daemon, these functions are implemented in [Provider Runtime crate](https://github.com/fiberplane/fiberplane-rs/blob/main/fiberplane-provider-protocol/fiberplane-provider-runtime/src/spec/mod.rs), should you want to inspect the source.
+Exported functions are functions that your provider implements, and that the runtime will call. In order to make those exported functions usable in Fiberplane, they must be annotated with `#[pdk_export]`.
 
-### `make_http_request`
+### `get_supported_query_types`
+
+It is advised to use the [Plugin Development Kit](https://docs.rs/fiberplane-pdk/latest/fiberplane_pdk/) rather than implementing this function manually. The [`pdk_query_types!`](https://docs.rs/fiberplane-pdk-macros/latest/fiberplane_pdk_macros/macro.pdk_query_types.html) macro can be used to generate this function for you. See the [tutorial](doc:create-a-provider) for more information about using the PDK.
 
 #### Signature
 
 ```rust
-pub async fn make_http_request(request: HttpRequest) -> Result<HttpResponse, HttpRequestError>;
+pub async fn get_supported_query_types(config: ProviderConfig) -> Vec<SupportedQueryType>;
 ```
 
 #### Description
 
-This is the most useful imported function, it allows the providers to make HTTP queries to remote hosts through the hosting runtime. A basic example of its usage can be seen in the tutorial to create a provider:
+Returns the query types supported by this provider. This function allows Studio to know upfront which formats will be supported, and which providers (and their query types) are eligible to be selected for certain use cases.
 
-```rust
-let base_url: Url = "https://jsonplaceholder.typicode.com".parse().unwrap();
-let url = base_url
-    .join("/users")
-    .unwrap();
+### `invoke2`
 
-let request = HttpRequest {
-    url,
-    headers: None,
-    method: HttpRequestMethod::Get,
-    body: None,
-};
+It is advised to use the [Plugin Development Kit](https://docs.rs/fiberplane-pdk/latest/fiberplane_pdk/) rather than implementing this function manually. The [`pdk_query_types!`](https://docs.rs/fiberplane-pdk-macros/latest/fiberplane_pdk_macros/macro.pdk_query_types.html) macro can be used to generate this function for you. See the [tutorial](doc:create-a-provider) for more information about using the PDK.
```

Comment:
```suggestion
It is advised to use the [Provider Development Kit](https://docs.rs/fiberplane-pdk/latest/fiberplane_pdk/) rather than implementing this function manually. The [`pdk_query_types!`](https://docs.rs/fiberplane-pdk-macros/latest/fiberplane_pdk_macros/macro.pdk_query_types.html) macro can be used to generate this function for you. See the [tutorial](doc:create-a-provider) for more information about using the PDK.
```